include/ros1_bridge/factory.hpp

```diff
@@ -33,6 +34,34 @@
 namespace ros1_bridge
 {
 
+static rclcpp::CallbackGroup::SharedPtr get_callback_group(
+  rclcpp::Node::SharedPtr ros2_node,
+  const std::string & topic_name = "")
+{
+  auto node_base = ros2_node->get_node_base_interface();
+  auto group = node_base->get_default_callback_group();
+  if (topic_name.empty()) {
+    // create a callback group with Reentrant for creating ros2 clients and services
+    static auto s_shared_callbackgroup =
+      ros2_node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
+    return s_shared_callbackgroup;
+  }
+
+  typedef std::map<std::string, rclcpp::CallbackGroup::SharedPtr> TopicCallbackGroupMap;
+  static TopicCallbackGroupMap s_topic_callbackgroups;
```

Comment:
instead of introducing `static auto s_shared_callbackgroup`, can we have everything in this map including `rclcpp::CallbackGroup::SharedPtr` for service and client?

Reply:
Maybe I can just use the empty string as the key